### ISSUE TITLE
no localtunnel response

### ISSUE THREAD

**Issue by dcsan** [saharmor/dalle-playground#94]
there seems to be an issue (on both my fork and your repo version)  that now I don't get a response from localtunnel:

wonder if this is a temporary colab glitch, or something changed in their infra to prevent this backdooring...

![image](https://user-images.githubusercontent.com/514002/175318709-78ff8aee-777d-4057-929c-dd918f7f4418.png)

**Comment by MarcDerhammer** [comment]
Same for me. My guess is Google cracking down.

**Comment by MarcDerhammer** [comment]
Nevermind, try it locally.  Seems localtunnel might be having service issues.  I was quick to jump to conclusions..

**Comment by alexis-de-la-torre** [comment]
Yes, this is a localtunnel issue: https://github.com/localtunnel/localtunnel/issues/501

**Comment by eyaler** [comment]
I just tested and both ngrok and argo are working for my in a non-related colab.

**Comment by MarcDerhammer** [comment]
I set up my instance to be exposed through `cloudflared` after reading https://github.com/localtunnel/localtunnel/issues/502 this.

Simply replace the `!npm install -g localtunnel` with this:
`!curl -L --output cloudflared.deb https://github.com/cloudflare/cloudflared/releases/latest/download/cloudflared-linux-amd64.deb && sudo dpkg -i cloudflared.deb && sudo cloudflared service install <insert your long token here>` 

the token can be retrieved after setting up an Access Tunnel within Cloudflare (Traffic -> Cloudflare Tunnel) and follow the instructions. 

Choose `Debian` and `64-bit` and copy the token from the `Install and run a connector` step

I already was using cloudflare for other things, so this was easy to setup using one of my existing domains.

After running the colab, Cloudflare should have detected the connection, and you can complete the configuration. 

Make sure the "Service" matches the port.  For example:  `http://localhost:8000` if running on port 8000

I did not try the "trycloudflare" setup, but if you're already using cloudflare this works fine.

**Comment by dcsan** [comment]
@trekkie1701c you can run a notebook locally too. 
any case no need for that just run the backend. however i get stuck here

```
Successfully built transformers dalle-mini
Failed to build tokenizers
ERROR: Could not build wheels for tokenizers, which is required to install pyproject.toml-based projects
```

**Comment by dcsan** [comment]
I got this connection running with ngrok 

```python
!pip install pyngrok

from threading import Thread
from pyngrok import ngrok
from time import sleep
import sys

# close existing
def restart_ngrok():
  tunnels = ngrok.get_tunnels()
  for tunnel in tunnels:
    ngrok.disconnect(tunnel.public_url)
  public_url = ngrok.connect(8000)
  print('ngrok public_url\n', public_url)
  ngrok_process = ngrok.get_ngrok_process()
  try:
      # Block until CTRL-C or some other terminating event
      ngrok_process.proc.wait()
  except KeyboardInterrupt:
      print(" Shutting down server.")
      ngrok.kill()

app_port = 8000
def app():
  print(f"Selected DALL-E Model - [{dalle_model}]")
  !python dalle-playground/backend/app.py --port {app_port} --model_version {dalle_model} --save_to_disk true --img_format jpeg --output_dir generations
  print(f'dalle starting on {app_port}')

if __name__=='__main__':
  print('start thread')
  t1 = Thread(target = app)
  a = t1.start()
  # !lt --port {app_port}
  restart_ngrok()  
else:
  print('not main')
```

**Comment by sjhamilton** [comment]
@dcsan Your solution seems to partially work. It generates a link for me, but when I try to use it on saharmor.me, the site is unable to find the DALL-E instance on the server. Any ideas or other info on what you did that might help me get it working?

**Comment by dcsan** [comment]
@sjhamilton I added a link to clarify when to start the ngrok:

tldr;
# make sure you don't restart the *blocking* ngrok service until after the app thread has started

**Comment by GreekPhysique** [comment]
I really don't get this. I put in your first set of code in my colab and  i got a link @dcsan 

But when i input the link into the front end website, it doesn't work.

This second jazz you wrote about the client side. I'm not sure what to do with that code

**Comment by PanJohnny** [comment]
@GreekPhysique same for me

**Comment by dcsan** [comment]
after you get the tunnel URL it can still take a few minutes til the model is ready. Not sure what else to say, it's working fine for me as a drop-in replacement. 

if you've tried to run it a few times, one thing you might have to do is go to ngrok's site and register for free, to get an auth token. Then you can set it with:

`  ngrok.set_auth_token("your-long-token-string-here")`

**Comment by GreekPhysique** [comment]
I'll try that.

**Comment by PanJohnny** [comment]
@dcsan isn't working for me, I looked into devtools and found this error: `backend_api.js:35 Mixed Content: The page at 'https://saharmor.me/dalle-playground/?backendUrl=http://SOMETHING.ngrok.io/' was loaded over HTTPS, but requested an insecure resource 'http://SOMETHING.ngrok.io/'. This request has been blocked; the content must be served over HTTPS.`

**Comment by PanJohnny** [comment]
Fixed it using HTTPS

**Comment by dcsan** [comment]
ok good one. yeah ngrok will give you both http and https URLs
i'm connecting from the server side, so it's not as relevant...

**Comment by GreekPhysique** [comment]
> Fixed it using HTTPS

@PanJohnny How did you fix it to HTTPS?

**Comment by GreekPhysique** [comment]
![image](https://user-images.githubusercontent.com/91274503/175661288-b0088e86-ead0-4709-8cb6-7a5149ae0eea.png)

I just copy that Ngrok tunnel url and paste it into the Dall-e playground front end website and i get

![image](https://user-images.githubusercontent.com/91274503/175661447-e9635b8e-a58e-4ab9-bff8-68e36f276373.png)

**Comment by GreekPhysique** [comment]
So much to learn still guys. Sooo much to learn!

**Comment by dcsan** [comment]
rather than `http://someurl` just add an extra 's' `httpS://someurl`

**Comment by GreekPhysique** [comment]
Tried that. I can access the link through my browser and i get {"success":true} with https


 but it just doesn't work on the DALL-E website.

**Comment by PanJohnny** [comment]
Worked after using my ngrok token

**Comment by GreekPhysique** [comment]
Ok i'll try that. where in the code do you set the token ? if you don't mind sharing @PanJohnny

**Comment by PanJohnny** [comment]
Here you go, worked for me
```python
!pip install pyngrok

from threading import Thread
from pyngrok import ngrok
from time import sleep
import sys

app_port = 8000

ngrok.set_auth_token("YOUR_TOKEN")

def restart_ngrok():
  # close existing
  tunnels = ngrok.get_tunnels()
  for tunnel in tunnels:
    ngrok.disconnect(tunnel.public_url)
  public_url = ngrok.connect(app_port)
  print('ngrok public_url\n', public_url)
  ngrok_process = ngrok.get_ngrok_process()
  try:
      # Block until CTRL-C or some other terminating event
      ngrok_process.proc.wait()
  except KeyboardInterrupt:
      print(" Shutting down server.")
      ngrok.kill()


def app():
  print(f"Selected DALL-E Model - [{dalle_model}]")
  !python dalle-playground/backend/app.py --port {app_port} --model_version {dalle_model} --save_to_disk true --img_format jpeg --output_dir generations
  print(f'dalle starting on {app_port}')

if __name__=='__main__':
  print('start thread')
  t1 = Thread(target = app)
  a = t1.start()
  # make sure you don't restart the *blocking* ngrok service until after the app thread has started
  # !lt --port {app_port}
  restart_ngrok()  
```

**Comment by PanJohnny** [comment]
When it prints out the url paste it into the website and instead of `http://` use `https://`. Also, you need to wait for this message: 
![image](https://user-images.githubusercontent.com/67548122/175665914-346a1953-6492-45d2-a487-4ad50568fbab.png)
That means that the model is loaded. Hope that helps 😃

**Comment by GreekPhysique** [comment]
Thanks very much! @PanJohnny. I'll give that a shot.

**Comment by GreekPhysique** [comment]
That didn't work. @PanJohnny . Did you pay for the ngrok service?

I'm think that might be the only difference here.
I'm getting a warning like so when i go to the link:

![image](https://user-images.githubusercontent.com/91274503/175667541-ea64461c-d910-4d5f-a5ad-fdc79cdfd327.png)


Thanks so much for your help btw

**Comment by GreekPhysique** [comment]
:(

**Comment by GreekPhysique** [comment]
I just suck

**Comment by ennauata** [comment]
same issue as @GreekPhysique :/

**Comment by MarcDerhammer** [comment]
Try to hit "visit site" or send the headers like the warning screen suggests.  Guessing your requests are returning the warning screen until you do that.

**Comment by PanJohnny** [comment]
If anyone wants. There is copy of the notebook with the ngrok stuff. [https://colab.research.google.com/drive/1qJ4pXxms70QCLdJ13raLzUPz5c_VKKGi?usp=sharing](https://colab.research.google.com/drive/1qJ4pXxms70QCLdJ13raLzUPz5c_VKKGi?usp=sharing)

**Comment by PanJohnny** [comment]
Hi made this gist with the ngrok stuff: https://gist.github.com/PanJohnny/03c68370206350f449ece2b21e088f21
